railties/lib/rails/test_unit/runner.rb

```diff
@@ -109,7 +109,7 @@ def escape_declarative_test_filter(filter)
             # transformations MUST BE idempotent.
             if filter.is_a?(String)
               if regexp_filter?(filter)
-                filter = filter.gsub(/\s+/, '[\s_]+')
+                filter = filter.gsub("\\ ", '[\s_]+').gsub(/\s+/, '[\s_]+')
```

Comment:
I do a lot of my test running using the [`m` gem](https://github.com/qrush/m). The regex I'm testing here comes from there. Specifically from [the `Regexp#escape` call on this line](https://github.com/qrush/m/blob/c86bf774f7daa5c3ea32e1f9597171fe30004e3e/lib/m/executor.rb#L20).

I realise it's not part of Rails so if the answer is we can't support it, I can try and find a way to support this in `m`. It just feels like it's something we should support since it's not like that gem is doing something really funky with the regex.

Reply:
I've pushed a commit that I think solves the problem.  Instead of wading through the weeds of regexps, we can simply join the original filter and the underscore-normalized filter.

Also, I discovered why the filter was being escaped over and over as you described in https://github.com/rails/rails/pull/47942#discussion_r1167676825: the same `options` instance is used for each call to `Rails::LineFiltering#run`, and we were modifying it in place each time.  I've fixed that, so `escape_declarative_test_filter` (now renamed `normalize_declarative_test_filter`) no longer has to be idempotent.

Reply:
> Instead of wading through the weeds of regexps, we can simply join the original filter and the underscore-normalized filter.

Damn. That's a great idea.

Reply:
This approach looks great to me 💯 

Thanks for your help!!